align/affineGap.go

```diff
@@ -1,259 +1,352 @@
-		} else if route[routeIdx].Op == k {
-			route[routeIdx].RunLength += 1
-		} else {
-			route = append(route, Cigar{RunLength: 1, Op: k})
-			routeIdx++
-		}
-		switch k {
-		case ColM:
-			k = trace[k][i][j]
-			i--
-			j--
-		case ColI:
-			k = trace[k][i][j]
-			j--
-		case ColD:
-			k = trace[k][i][j]
-			i--
-		default:
-			log.Fatalf("Error: unexpected traceback")
+		for idx := range trace_prep_j[0] {
+			trace_prep_j[k][idx] = make([]int64, firstSeqLen+1)
 		}
 	}
-	reverseCigar(route)
-	return maxScore, route
+	return mRowCurrent, mRowPrevious, mColumn, trace_prep_i, trace_prep_j
 }
 
-func expandCigarRunLength(route []Cigar, chunkSize int64) {
-	for i := range route {
-		route[i].RunLength *= chunkSize
+//"Step 0"
+//make data structures for tracing
+func initAffineTrace(firstSeqLen int, secondSeqLen int, checkersize_i int, checkersize_j int) [][][]ColType {
+	trace_size_i := numbers.Min(firstSeqLen, checkersize_i) //make trace a matrix of size checkersize_i*checkersize_j, unless alpha or beta are shorter, in which case there is no need to allocate a full checkersize of memory
+	trace_size_j := numbers.Min(secondSeqLen, checkersize_j)
+	trace := make([][][]ColType, 3)
+	for k := range trace { //k ranges through 3 numbers (0,1,2)
+		trace[k] = make([][]ColType, trace_size_i)
+		for i := range trace[0] {
+			trace[k][i] = make([]ColType, trace_size_j)
+		}
 	}
+	return trace
 }
 
+//This version of AffineGap has a fixed checkersize of 10000*10000
 func AffineGap(alpha []dna.Base, beta []dna.Base, scores [][]int64, gapOpen int64, gapExtend int64) (int64, []Cigar) {
-	mRowCurrent, mRowPrevious, mColumn, trace := initAffineScoringAndTrace(len(alpha), len(beta))
-	for i := 0; i < mColumn; i++ {
-		for j := range mRowCurrent[0] {
-			if i == 0 && j == 0 {
-				mRowCurrent[0][j] = 0
-				mRowCurrent[1][j] = gapOpen
-				mRowCurrent[2][j] = gapOpen
-			} else if i == 0 {
-				mRowCurrent[0][j] = veryNegNum
-				mRowCurrent[1][j] = gapExtend + mRowCurrent[1][j-1]
-				trace[1][i][j] = ColI /*new*/
-				mRowCurrent[2][j] = veryNegNum
-			} else if j == 0 {
-				mRowCurrent[0][j] = veryNegNum
-				mRowCurrent[1][j] = veryNegNum
-				mRowCurrent[2][j] = gapExtend + mRowPrevious[2][j]
-				trace[2][i][j] = ColD /*new*/
-			} else {
-				mRowCurrent[0][j], trace[0][i][j] = tripleMaxTrace(scores[alpha[i-1]][beta[j-1]]+mRowPrevious[0][j-1], scores[alpha[i-1]][beta[j-1]]+mRowPrevious[1][j-1], scores[alpha[i-1]][beta[j-1]]+mRowPrevious[2][j-1])
-				mRowCurrent[1][j], trace[1][i][j] = tripleMaxTrace(gapOpen+gapExtend+mRowCurrent[0][j-1], gapExtend+mRowCurrent[1][j-1], gapOpen+gapExtend+mRowCurrent[2][j-1])
-				mRowCurrent[2][j], trace[2][i][j] = tripleMaxTrace(gapOpen+gapExtend+mRowPrevious[0][j], gapOpen+gapExtend+mRowPrevious[1][j], gapExtend+mRowPrevious[2][j])
-			}
-		}
-		if i < mColumn-1 {
-			mRowPrevious, mRowCurrent = mRowCurrent, mRowPrevious
+	var checkersize_i, checkersize_j int
+	checkersize_i = 10000
+	checkersize_j = 10000
+
```

Comment:
I think it is good to have this function, but I would replace it with a call to:
AffineGap_customizeCheckersize
after setting the checkersize to be 10000.  This way you don't have code duplication.
For my comments, I am going to skip ahead to the custom checkersize function.

Reply:
Just fixed this